tests/xharness/Utilities/Helpers.cs

```diff
@@ -25,5 +26,10 @@ public static Guid GenerateStableGuid (string seed = null)
 			}
 			return new Guid (bytes);
 		}
+
+		public static string Timestamp => $"{DateTime.Now:yyyyMMdd_HHmmss}";
+
+		[DllImport ("/usr/lib/libc.dylib")]
+		extern internal static IntPtr Ttyname (int filedes);
```

Comment:
```suggestion
		extern internal static IntPtr ttyname (int filedes);
```

Reply:
And this would be better exposed with a managed wrapper:

```csharp
[DllImport ("/usr/lib/libc.dylib")]
extern static IntPtr ttyname (int filedes);

internal static string GetTerminalName (int filedescriptor)
{
     return Marshal.PtrToStringAuto (ttyname (filedescriptor));
}
```